### PR TITLE
PERF: Groupby.idxmax

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @phofl this is currently failing locally and could use a fresh pair of eyes.

### CODE REVIEW

**Comment by phofl** [comment]
I'll try to take a look over the weekend

**Comment by jbrockmendel** [comment]
@phofl im mothballing this to clear the queue, just a ping so this doesn't fall off your radar entirely

**Comment by phofl** [comment]
I think the main problem is that ``_wrap_aggregated_output`` isn't working for unordered categories that are empty. All other reductions that are tested are raising TypeErrors, while idxmax and idxmin return the wrong result. Any ideas what we could do there?

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1138,13 +1138,51 @@ def nsmallest(
     def idxmin(
         self, axis: Axis | lib.NoDefault = lib.no_default, skipna: bool = True
     ) -> Series:
+        if axis is lib.no_default:
+            alt = lambda x: Series(x).argmin(skipna=skipna)
+            with com.temp_setattr(self, "observed", True):
+                argmin = self._cython_agg_general("argmin", alt=alt, skipna=skipna)
```
Comment (rhshadrach) [inline pandas/core/groupby/generic.py:1144]: There is a bit of an oddity with how missing observations are handled. When there are multiple groupings, we do not include the unobserved categories in e.g. `grouper.result_index` and fill in any unobserved ones in `_wrap_aggregated_output`. However, if there is just a single grouping we _do_ include the unobserved groupings in e.g. `grouper.result_index` and so we don't fill them in later on. This makes this approach not work for some cases of categoricals.

I plan to look into making it so we never include the unobserved categories until `_wrap_aggreagted_output` in the single grouping case. If we can make that work, this approach would work.

**Pull request closed without merging**